src/lazyload-image.ts

```diff
@@ -9,12 +9,22 @@ import { Observable } from 'rxjs/Observable';
 import { getScrollListener } from './scroll-listener';
 import { Rect } from './rect';
 
-export function isVisible(element: HTMLElement, threshold = 0, _window = window) {
+export function isVisible(element: HTMLElement, threshold = 0, _window: Window, scrollContainer?: HTMLElement) {
     const elementBounds = Rect.fromElement(element);
     const windowBounds = Rect.fromWindow(_window);
     elementBounds.inflate(threshold);
-    
-    return elementBounds.intersectsWith(windowBounds);
+
+    if (scrollContainer) {
+        const scrollContainerBounds = Rect.fromElement(scrollContainer);
+        if (scrollContainerBounds.intersectsWith(windowBounds)) {
+            const intersection = scrollContainerBounds.getIntersectionWith(windowBounds);
+            return elementBounds.intersectsWith(intersection);
+        } else {
+            return false;
+        }
+    } else {
+        return elementBounds.intersectsWith(windowBounds);
+    }
```

Comment:
What's your preferred code convention for these kind of `else` statements? Do you prefer it like it is, or like this:
```javascript
if (scrollContainer) {
    const scrollContainerBounds = Rect.fromElement(scrollContainer);
    if (scrollContainerBounds.intersectsWith(windowBounds)) {
        const intersection = scrollContainerBounds.getIntersectionWith(windowBounds);
        return elementBounds.intersectsWith(intersection);
    }
    return false;
}
return elementBounds.intersectsWith(windowBounds);
```